### PR TITLE
proxmark3 refactoring command line parameters

### PR DESCRIPTION
**Pull request by merlokk**
small refactoring + added feature to wait com port on proxmark start

### CODE REVIEW

**Comment by merlokk** [comment]
maybe ill add -c parameter to start proxmark command from commandline

**Comment by merlokk** [comment]
works.
It needs to check. Many changes in parts into `ifdef` code(....

sample for win:
proxmark3.exe com6 -w -c "hf mf nested 1 *"
proxmark3.exe com6 -w -c "hf mf chk 1* ?"

**Comment by merlokk** [comment]
it seems) all is ok

```
proxmark3.exe com6 -w -c "hf mf nested 1 *"
proxmark3.exe com6 -c "hf mf chk 1* ?"
proxmark3.exe com6 -l hf_read
```
ready for test and merge

**Comment by iceman1001** [comment]
Nice changes, Great changes,  some I recognise and some interesting new ones.

However,  it would be great if you could make small, simple PR. Which is easy to test out and **not** add more commits as it goes.  Otherwise it is gonna take even longer to test the PR.

To conclude,  please, 
- Don't make a PR if your change is not ready to be commited.
- Don't add commits over and over. 
- test your own code before making PR

**Comment by merlokk** [comment]
ok) now it ready. 
I will publish next PR after i have tested changes.

File: client/proxmark3.c [hunk 1]
```diff
@@ -254,44 +369,35 @@ int main(int argc, char* argv[]) {
...
-		{
-			printf("Output will be flushed after every print.\n");
-			flushAfterWrite = 1;
-		}
-		else
-		script_cmds_file = argv[2];
+	
+	// dont execute comamnds in offline mode
+	if (offline && (executeCommand || script_cmds_file)) {
+		printf("ERROR: can't execute in offline mode.\n");
```
Comment (pwpiwi) [inline client/proxmark3.c]: Please remove. There are commands which can be executed in offline mode.

Comment (merlokk) [reply client/proxmark3.c]: deleted

**Comment by merlokk** [comment]
dont execute commands. deleted

**Comment by iceman1001** [comment]
question,  now that you added the "-c" param,  what is actually needed for us to use pipe:ing ?
like this:
echo "hf 14a reader" | proxmark3 -w   

I believe this will make the pm3 client much friendler to use in shellscripts, or other scripts

**Comment by merlokk** [comment]
https://stackoverflow.com/questions/15467164/why-does-readline-echo-stdin-to-stdout
The readline library is solely intended for interactive use
ill make it by getline()...

**Comment by iceman1001** [comment]
Pipe works,  on win/linux/osx ?  nicely done!

**Comment by merlokk** [comment]
works. try
printf "hf 14a reader\nhf mf rdbl 0 a ffffffffffff\n" | proxmark3 com6  -w

**Comment by pwpiwi** [comment]
Works fine on Windows and Kali. Passes Travis Tests.

Please add a line to CHANGELOG.md

**Comment by merlokk** [comment]
added

**Pull request merged**